### PR TITLE
master lpd 31321 B

### PR DESCRIPTION
**Pull request by kyle-miho**
Forwarded from : https://github.com/liferay-devtools/liferay-portal/pull/272
https://liferay.atlassian.net/browse/LPD-31321
@nicolas-sss I made it simpler using your suggestions, still need to do some replacements if classes implemented clayFDS... or commerceFDS

The auto-renaming for UpgradeJavaCheck is simple, I'm not sure if we want to update the capitilization logic so it doesnt ugprade to things like fDS instead of fds, or if its not worth the effort

### CODE REVIEW

**Comment by kyle-miho** [comment]
ci:test:sf

File: modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaFDSDataProviderCheck.java [hunk 1]
```diff
@@ -172,14 +184,57 @@ private String _reorderGetItemsCount(String methodCall) {
...
+	private String _updateImplementedClass(String content) {
+		Matcher matcher = _implementedClassPattern.matcher(content);
+
+		if (matcher.find()) {
+			return StringUtil.replace(
+				content, matcher.group(), matcher.group(1) + "FDSDataProvider");
+		}
+
+		return content;
+	}
```
Comment (nicolas-sss) [inline modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaFDSDataProviderCheck.java:196]: Hi @kyle-miho, since the imports.txt will be executed before, this will never happen, will it?

File: modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/upgrade/UpgradeJavaClayDataSetDataProviderCheck.testjava [hunk 2]
```diff
@@ -5,6 +5,10 @@
 
 package com.liferay.source.formatter.dependencies.upgrade;
 
+import com.liferay.frontend.taglib.clay.data.set.provider.ClayDataSetDataProvider;
+import com.liferay.frontend.taglib.clay.data.Filter;
+import com.liferay.frontend.taglib.clay.data.Pagination;
```
Comment (nicolas-sss) [inline modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/upgrade/UpgradeJavaClayDataSetDataProviderCheck.testjava:10]: @kyle-miho why did you add these imports? 
If isn't necessary to the logic, we can keep it simple and remove it, what do you think?

File: modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaFDSDataProviderCheck.java [hunk 3]
```diff
@@ -33,6 +34,14 @@ protected String format(
 		List<String> implementedClassNames =
 			javaClass.getImplementedClassNames();
 
+		if (implementedClassNames.contains("FDSDataProvider")) {
+			content = _updateProviderKey(content);
+
+			content = _updateImplementedClass(content);
```
Comment (nicolas-sss) [inline modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaFDSDataProviderCheck.java:40]: If the UpgradeJavaCheck has already been executed, the class will be like this:

Before
```
 public class UpgradeJavaClayDataSetDataProviderCheck implements ClayDataSetDataProvider<UpgradeJava>
```

After
```
 public class UpgradeJavaClayDataSetDataProviderCheck implements FDSDataProvider<UpgradeJava>
```
Because of that, it will be validated by this if statement.

If that is the case, the following line is doing what?

`content = _updateImplementedClass(content);
`

**Review by nicolas-sss — CHANGES_REQUESTED** [review]

File: modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/upgrade/UpgradeJavaClayDataSetDataProviderCheck.testjava (hunk 2 above)
Comment (kyle-miho) [reply modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/upgrade/UpgradeJavaClayDataSetDataProviderCheck.testjava:10]: @nicolas-sss the imports.txt replacements depend on this, but i think its better to remove it, so we can still perform the upgrade and fix their missing imports if they dont include these, ill just add the imports in newImports if not already present to account for this

Comment (kyle-miho) [reply modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/upgrade/UpgradeJavaClayDataSetDataProviderCheck.testjava:10]: @nicolas-sss this is technically required since we are using imports.txt / UpgradeJavaCheck to perform a large bulk of the upgrades like you told me, we need those imports to exist otherwise it wont perform the replacements for all of its usages.

**Comment by kyle-miho** [comment]
@nicolas-sss I put a comment above why those imports are needed in the before cases, its because we are utilizing the imports.txt/UpgradeJavaCheck method of upgrading, and thus require the imports as a point of reference to upgrade the imports themselves as well as its usages. I fixed the un-needed implementation class upgrade now since it is technically also covered by UpgradeJavaCheck as well

**Review by nicolas-sss — APPROVED** [review]

**Review by drewbrokke — APPROVED** [review]

**Comment by kyle-miho** [comment]
ci:test:hotel

**Comment by kyle-miho** [comment]
https://github.com/brianchandotcom/liferay-portal/pull/155783
